packages/ui-components/src/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useCallback, useMemo } from "react";
+import classNames from "classnames/bind";
+
+import styles from "./Avatar.module.scss";
+import objectToClassnames from "../utils/objectToClassnames";
+
+export interface AvatarProps {
+  children?: string;
+  size?: AvatarSize;
+  intent?: AvatarIntent;
+  disabled?: boolean;
+  selectable?: boolean;
+  onClick?: () => void;
+}
+
+export type AvatarSize = "16" | "24" | "32" | "40";
+export type AvatarIntent = "default" | "info" | "warning" | "danger";
+
+const cx = classNames.bind(styles);
+
+const Avatar: React.FC<AvatarProps> = ({
+  children,
+  intent,
+  size,
+  disabled,
+  selectable,
+  onClick,
+}) => {
+  const classnames = useMemo(
+    () =>
+      cx("avatar", objectToClassnames({ size }), {
+        disabled,
+        selectable: !disabled && selectable,
+        [`intent-${intent}`]: !disabled,
+      }),
+    [intent, size, disabled, selectable],
+  );
+
+  const onClickHandler = useCallback(() => {
+    if (!disabled && onClick) {
+      onClick();
+    }
+  }, [onClick, disabled]);
+
+  return (
+    <div className={classnames} onClick={onClickHandler}>
+      {children}
+    </div>
+  );
+};
+
+Avatar.defaultProps = {
+  size: "40",
+  intent: "default",
+  selectable: false,
+  disabled: false,
+};
```

Comment:
As far as setting default props, I kind of like this pattern here but I know in other places we've specified them in function signature itself. @laurenbarker do you have an opinion on this at all?

Reply:
`defaultProps` is deprecated for function components (https://github.com/facebook/react/pull/16210) which is why we decided to go with [default parameters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters) elsewhere.